tools/pythonpkg/tests/fast/udf/test_scalar_arrow.py

```diff
@@ -124,6 +124,30 @@ def return_too_many(col):
         with pytest.raises(duckdb.InvalidInputException, match='Returned pyarrow table should have 1 tuples, found 5'):
             res = con.sql("""select too_many_tuples(5)""").fetchall()
 
+    def test_arrow_side_effects(self, duckdb_cursor):
+        import random as r
+
+        def random_arrow(x):
+            if 'data' not in random_arrow.__dict__:
+                random_arrow.__dict__["data"] = 0
```

Comment:
Is there a reason you're doing this instead of just 


```suggestion
                random_arrow.data = 0
```

and using `hasattr()`?